### ISSUE TITLE
User file with PTHP has negative heating electricity

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#6396]
Issue overview
--------------
User file with PTHP shows a negative value for heating electricity.  Further investigation shows that DX heating coil ratings show a negative value for low temperature heating capacity:
```
! <DX Heating Coil Standard Rating Information>, Component Type, Component Name, High Temperature Heating (net) Rating Capacity {W}, Low Temperature Heating (net) Rating Capacity {W}, HSPF {Btu/W-h}, Region Number
 DX Heating Coil Standard Rating Information, Coil:Heating:DX:SingleSpeed, THERMAL ZONE: SPACE 101_WEST PTHP DX HEATING COIL, 115912.6, -16638.0, 5.40, 4
```
This points to the Heating Capacity Function of Temperature Curve as the problem.
```
Curve:Cubic,
  Thermal Zone: Space 102_South PTHP Heating CAPFT, !- Name
  0.876825,                               !- Coefficient1 Constant
  -0.002955,                              !- Coefficient2 x
  5.8e-005,                               !- Coefficient3 x**2
  0.025335,                               !- Coefficient4 x**3
  -5,                                     !- Minimum Value of x {BasedOnField A2}
  25;                                     !- Maximum Value of x {BasedOnField A2}

```
The DX heating coil object (or any coil for that matter) should reject a negative heating capacity and/or negative heating electricity.  In the case of a negative heating capacity, one option is to reset to zero and throw a recurring warning.  Negative electric power should probably be a severe/fatal error.

*Update*
The heating CAPFT curve is part of the problem - it contributes to the negative low-temp rated capacity, but adding outputs for curve values shows that this curve is not going negative during the simulation.  The negative power problem is from the Energy Input Ratio Function of Temperature Curve, "Curve Cubic 16". This curve is negative above 4.5C, and at 25C, the curve value is -140.
```
  Curve:Cubic,
    Curve Cubic 16,          !- Name
    0.704658,                !- Coefficient1 Constant
    0.008767,                !- Coefficient2 x
    0.000625,                !- Coefficient3 x**2
    -0.009037,               !- Coefficient4 x**3
    -5,                      !- Minimum Value of x
    25;                      !- Maximum Value of x
```

### Workaround
Fix the curve or add fields for Minimum Curve Output (and Maximum Curve Output) to keep the curve result within reasonable limits.

### Details
Some additional details for this issue (if relevant):
 - Version of EnergyPlus v8.8.0
 - Helpdesk ticket number 12399

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added EnergyPlusDevSupport\DefectFiles
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by Nigusse** [comment]
@mjwitte Adding "Output:Diagnostics, DisplayExtraWarnings;" shows most of the curves have problem including those which are positive values. I suspect the curves are not correctly generated. 

   ** Warning ** GetDXCoils: Coil:Heating:DX:SingleSpeed="THERMAL ZONE: SPACE 101_WEST PTHP DX HEATING COIL" curve values
   **   ~~~   ** ... Heating Capacity Function of Temperature Curve Name = THERMAL ZONE: SPACE 101_WEST PTHP HEATING CAPFT output is not equal to 1.0 (+ or - 10%) at rated conditions.
   **   ~~~   ** ... Curve output at rated conditions = 15.500
   ** Warning ** GetDXCoils: Coil:Heating:DX:SingleSpeed="THERMAL ZONE: SPACE 101_WEST PTHP DX HEATING COIL" curve values
   **   ~~~   ** ... Energy Input Ratio Function of Temperature Curve Name = CURVE CUBIC 16 output is not equal to 1.0 (+ or - 10%) at rated conditions.
   **   ~~~   ** ... Curve output at rated conditions = -4.402
   ** Warning ** GetDXCoils: Coil:Heating:DX:SingleSpeed="THERMAL ZONE: SPACE 102_SOUTH PTHP DX HEATING COIL" curve values
   **   ~~~   ** ... Heating Capacity Function of Temperature Curve Name = THERMAL ZONE: SPACE 102_SOUTH PTHP HEATING CAPFT output is not equal to 1.0 (+ or - 10%) at rated conditions.
   **   ~~~   ** ... Curve output at rated conditions = 15.500
   ** Warning ** GetDXCoils: Coil:Heating:DX:SingleSpeed="THERMAL ZONE: SPACE 102_SOUTH PTHP DX HEATING COIL" curve values
   **   ~~~   ** ... Energy Input Ratio Function of Temperature Curve Name = CURVE CUBIC 19 output is not equal to 1.0 (+ or - 10%) at rated conditions.
   **   ~~~   ** ... Curve output at rated conditions = -4.402
   ** Warning ** GetDXCoils: Coil:Heating:DX:SingleSpeed="THERMAL ZONE: SPACE 103_EAST PTHP DX HEATING COIL" curve values
   **   ~~~   ** ... Heating Capacity Function of Temperature Curve Name = THERMAL ZONE: SPACE 103_EAST PTHP HEATING CAPFT output is not equal to 1.0 (+ or - 10%) at rated conditions.
   **   ~~~   ** ... Curve output at rated conditions = 15.500
   ** Warning ** GetDXCoils: Coil:Heating:DX:SingleSpeed="THERMAL ZONE: SPACE 103_EAST PTHP DX HEATING COIL" curve values
   **   ~~~   ** ... Energy Input Ratio Function of Temperature Curve Name = CURVE CUBIC 18 output is not equal to 1.0 (+ or - 10%) at rated conditions.
   **   ~~~   ** ... Curve output at rated conditions = -4.402

**Comment by Myoldmopar** [comment]
Closed via #6441
